examples/plot_04_competing_risks.py

```diff
@@ -0,0 +1,167 @@
+    palette="colorblind",
+)
+plt.show()
+
+# %%
+from sklearn.model_selection import train_test_split
+from hazardous import SurvivalBoost
+
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+survival_boost = SurvivalBoost(n_iter=10, show_progressbar=False).fit(X_train, y_train)
+survival_boost
+
+# %%
+
+incidence_curves = survival_boost.predict_cumulative_incidence(
+    X_test,
+    times=None,
+)
+
+survival_curves = incidence_curves[:, 0]  # survival function S(t)
+
+# %%
+# Let's plot the estimated survival function and the incidence functions for
+# some patients.We will also plot a symbol at the time of death or censoring.
+# The symbol will be a skull and crossbones for the event that occurred,
+# a cross for meaning that an other event of interest has happened, and a
+# question mark for the censoring event.
+
+import matplotlib.pyplot as plt
+
+fig, ax = plt.subplots(nrows=1, ncols=4, figsize=(15, 4))
+sns.despine()
+patient_ids_to_plot = [0, 1, 2, 3]
+
+for event_id in range(n_events + 1):
+    for idx in patient_ids_to_plot:
+        ax[event_id].plot(
+            survival_boost.time_grid_,
+            incidence_curves[idx, event_id],
+            label=f"Patient {idx}",
+        )
+
+        # plot symbols for death or censoring
+        event = y_test.iloc[idx]["event"]
+        duration = y_test.iloc[idx]["duration"]
+
+        # find the index of time closest to duration
+        jdx = np.searchsorted(survival_boost.time_grid_, duration)
+        smiley = "\N{SKULL AND CROSSBONES}" if event == event_id else "✖"
+        smiley = "\N{QUESTION MARK}" if event == 0 else smiley
+        ax[event_id].text(
+            duration,
+            incidence_curves[idx, event_id, jdx],
+            smiley,
+            fontsize=20,
+            color=ax[event_id].lines[idx].get_color(),
+        )
+
+    ax[event_id].set_xlabel("Time")
+    if event_id == 0:
+        ax[event_id].set_title("Survival Function")
+    else:
+        ax[event_id].set_title(f"Incidence Fucntion, Event {event_id}")
+
+
+ax[0].set_ylabel("Survival Probability")
+ax[1].set_ylabel("Incidence Probabilities")
+ax[-1].legend(loc="lower right")
+plt.legend(bbox_to_anchor=(1.05, 1), loc="upper left")
+
+plt.show()
+
+# %%
+# Let's compute the incidence functions and the survival function for a given patient
+# knowing its censoring time.
+# Given a patient that has been censored at a given time, we do know that she/he has not
+# experienced any event before this time. We can use this information to compute the
+# survival function and the incidence functions for the whole period.
+five_first_censored_patients = X_test[y_test["event"] == 0].iloc[:5]
+censoring_times = y_test[y_test["event"] == 0]["duration"].iloc[:5]
+
+censoring_function_knowing_censoring_time = survival_boost.predict_incidence_after_s(
+    five_first_censored_patients, censoring_times.values
+)
+
+
+# %%
+fig, ax = plt.subplots(nrows=1, ncols=4, figsize=(15, 4))
```

Comment:
1. Could you display the plots column-wise rather than row-wise? This way they would be easier to see.
2. Could you also put the plotting logic in a function (placed in this example) so that we only see the whole logic once?